### ISSUE TITLE
Wait until Message Service modal is closed

### ISSUE THREAD

**Issue by mdmontesinos** [Megabit/Blazorise#4178]
I was wondering why different message types do not behave the same way as the Confirm does.

For example, with a code like this:
```c#
await messageService.Info("First Message");
await messageService.Info("Second Message");
Console.WriteLine("Console Message");
```
You immediately get the second message service modal and the console message displayed (the first one never appears), without truly waiting until the messages are closed.

However, if you use the Confirm one instead:
```c#
await messageService.Confirm("First Message");
await messageService.Confirm("Second Message");
Console.WriteLine("Console Message");
```
You get them displayed step by step, first displaying "First Message" modal and, upon closing it, the second one is displayed, and when it's closed, then the console message appears. Therefore, there's a real "awaiting" process.

I know with the Confirm one you get the boolean, and therefore you must wait until you get the value, but why don't the other modals also keep the process waiting until the OK button is clicked?

**Comment by stsrki** [comment]
It was always an idea to show only one message to the user. And if called multiple times, it will always take the last message received. Why Confirm behaves differently. Mainly because, unlike other messages, it internally uses `TaskCompletionSource`, which waits for the response, but even so, it still handles only one message internally.

**Comment by mdmontesinos** [comment]
Well, the two messages was just an example to represent the problem. 

In my case, the issue is that I need the app to wait until the OK Button is clicked to avoid redirecting the user beforehand (but the redirect does not depend on a choice, so I don't want a Confirm).

```c#
await messageService.Info("Some message");
navigationManager.NavigateTo("/somepage");
```
Currently, the Message is displayed but I get instantly redirected to the other page, before waiting for the user to click the OK Button.

**Comment by mdmontesinos** [comment]
Cool, I'll try that once I update to the newest version.

**Comment by David-Moreira** [comment]
@stsrki solution seems the way to go. And it's pretty simple.

If you want more control you can try checking our new `ModalProvider` component. 
You should be able to handle that logic by instantiating the first message modal and then bind to the `Closed` event.

Altough it might be overkill if it's all just simple messages. :)

@stsrki are you thinking on improving anything here, or may we go ahead and close this issue?

**Comment by stsrki** [comment]
Well, I think it might be a good approach to keep a stack of opened messages/modals. For example if we call Info multiple times every message would show its own modal on top of each other. But waiting for them is still not a good idea, IMO.

**Comment by David-Moreira** [comment]
Right, that'd be the behaviour I expect too by default. Them stacked.

**Comment by stsrki** [comment]
That's already what we do with ModalService.

**Comment by mdmontesinos** [comment]
To be honest, the solution provided by @stsrki is kind of "dirty", in the sense that I need to hide a button to get the desired functionality. 

The other solution provided by @David-Moreira using ModalProvider, as he said, it's a bit overkill for a simple message.

I believe that an interesting solution would be having a parameter in MessageOptions that controls whether the message should be blocking or not, defaulting to true for Confirm, and false for the others. However, you mentioned that Info should be non-blocking , so I don't know if there's a specific reason that I'm missing.

**Comment by David-Moreira** [comment]
I guess that's just what it feels like. If we had a `HideCancelButton` parameter then it wouldn't feel as much of "dirty" even though the end result is the same.

**Comment by stsrki** [comment]
> However, you mentioned that Info should be non-blocking , so I don't know if there's a specific reason that I'm missing.

It shouldn't block the flow so that it is fire and forget. The same thing when you log the message with `ILogger`. That's the idea behind the MessageService, to give an indication of something to the user while the background process is still running.

Now, we might consider adding an option to wait for the user action. It's not set in stone.

**Comment by mdmontesinos** [comment]
I've tried the solution that uses the custom css class, but it's not working as expected. I added the custom class you suggested to app.css:
```css
.hide-button {
    display: none;
}
```
Then added to class name to CancelButtonClass:
```c#
await messageService.Confirm(texto, "Resultados Examen", options =>
                {
                    options.ConfirmButtonText = "Volver a la sala de vídeos";
                    options.CancelButtonClass = "hide-button";
                });
```
However, this is the modal that appears:
![image](https://user-images.githubusercontent.com/90258222/194828902-de9950bb-fd1b-4385-8108-46f33014dd6a.png)
As you can see the button is present, and it does show that the class is applied. Could it be a problem with the display property being overriden by another class?

**Comment by stsrki** [comment]
The order of CSS is always problematic, so it is possible that some later CSS style overwrites it. Cas you add `display: none !important;` and see if it works?

**Comment by mdmontesinos** [comment]
Yes, that worked, thanks!

**Comment by stsrki** [comment]
Working since 2.2.x. Closing now.
